### PR TITLE
AposAttachments UI & story

### PR DESCRIPTION
**Pull request by stuartromanek**

### CODE REVIEW

**Comment by stuartromanek** [comment]
This is only a partial storybook implementation .. there are a number of things that need to be cleaned up about Slats to make a unified experience across the components that use SlatList.. After that happens I can do a pt 2 here where we emit events up when changing the field value (removing items). We also need to decide on a file drop approach.

**Comment by abea** [comment]
Linties.

**Review by falkodev — APPROVED** [review]

File: modules/@apostrophecms/schema/ui/apos/components/AposAttachment.vue [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          />
+        </div>
+      </div>
+    </template>
+  </AposInputWrapper>
+</template>
+
+<script>
+import AposInputMixin from '../mixins/AposInputMixin.js';
+import AposSlatList from '../../../../ui/ui/apos/components/AposSlatList';
```
Comment (abea) [inline modules/@apostrophecms/schema/ui/apos/components/AposAttachment.vue:25]: We should use the `Modules` webpack alias for both of these.

File: modules/@apostrophecms/ui/ui/apos/components/AposSlat.vue [hunk 2]
```diff
@@ -18,11 +18,24 @@
...
+            {{ item.ext }}
+          </span>
+        </div>
         <div class="apos-slat__label">
           {{ item.title }}
         </div>
       </div>
       <div class="apos-slat__secondary">
+        <div class="apos-slat__size" v-if="item.size">{{ item.size }}</div>
         <close-icon
```
Comment (abea) [inline modules/@apostrophecms/ui/ui/apos/components/AposSlat.vue:39]: This close icon needs to be a button. You could use the `AposButton` and the icon props, or wrap this directly here.

File: modules/@apostrophecms/ui/ui/apos/components/AposSlat.vue [hunk 3]
```diff
@@ -150,4 +167,55 @@ export default {
...
+  .apos-slat__extension--numbers,
+  .apos-slat__extension--csv {
+    background-color: var(--a-spreadsheet);
+  }
+
+  .apos-slat__extension--key,
+  .apos-slat__extension--ppt,
+  .apos-slat__extension--pptx {
+    background-color: var(--a-presentation);
+    // color: var(--a-primary-text);
```
Comment (abea) [inline modules/@apostrophecms/ui/ui/apos/components/AposSlat.vue:209]: comment here.

File: modules/@apostrophecms/ui/ui/apos/components/AposSlatList.vue [hunk 4]
```diff
@@ -90,8 +91,11 @@ export default {
       this.engaged = null;
     },
     remove(item, focusNext) {
+      console.log('removing');
```
Comment (abea) [inline modules/@apostrophecms/ui/ui/apos/components/AposSlatList.vue:94]: Three logs here.

**Review by abea — CHANGES_REQUESTED** [review]
When I remove one to be less than the max, the "Limit Reached" message doesn't disappear. It probably needs to refresh that on the `remove` event.

File: modules/@apostrophecms/ui/ui/apos/components/AposSlat.vue (hunk 2 above)
Comment (stuartromanek) [reply modules/@apostrophecms/ui/ui/apos/components/AposSlat.vue]: made a new `inline` button modifier that is basically an AposButton with no chrome

**Comment by stuartromanek** [comment]
@abea better limit watching

**Review by abea — APPROVED** [review]

**Pull request merged**